compiler/rustc_passes/src/reachable.rs

```diff
@@ -318,14 +318,17 @@ impl<'tcx> ReachableContext<'tcx> {
                     ));
                     self.visit(instance.args);
                 }
-                GlobalAlloc::VTable(ty, trait_ref) => {
+                GlobalAlloc::VTable(ty, vtable) => {
                     self.visit(ty);
                     // Manually visit to actually see the trait's `DefId`. Type visitors won't see it
-                    if let Some(trait_ref) = trait_ref {
+                    if let Some(trait_ref) = vtable.principal() {
                         let ExistentialTraitRef { def_id, args } = trait_ref.skip_binder();
                         self.visit_def_id(def_id, "", &"");
                         self.visit(args);
                     }
+
+                    // FIXME: are we just skipping the auto traits here?
+                    // What is this reachability *for*?
```

Comment:
There's a very detailed comment at the top of this file explaining what it is for.